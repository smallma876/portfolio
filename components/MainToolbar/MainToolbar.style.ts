import styled from "styled-components";

const Toolbar = styled.div`
  background: #244072;
  height: 3.5rem;
  color: #fff;
  padding: 0 1rem;
`;

interface MenuLateralProps {
  isShow: boolean;
}

const MenuLateral = styled.div<MenuLateralProps>`
  position: fixed;
  top: 0;
  left: ${/*istanbul ignore next */ (props) => (props.isShow ? "0" : "-10rem")};
  bottom: 0;
  width: 10rem;
  background: #04293a;
  transition: left 0.3s ease-out;
  min-width: 10rem;
`;

const IconCloseWrapper = styled.div`
  display: flex;
  align-items: center;
  margin: 0;
  padding: 0 1rem;
  border: 0;
  background: transparent;
  height: 3.5rem;
`;

export const MainToolbarStyles = {
  Toolbar,
  MenuLateral,
  IconCloseWrapper,
};
