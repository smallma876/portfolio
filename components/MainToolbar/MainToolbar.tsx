import Image from "next/image";
import React, { useEffect, useState } from "react";
import { Box, Flexbox } from "../../shared/components/Flexbox/Flexbox";
import { useMediaQuery } from "../../utils/useMediaQuery";
import NavbarRoutes from "../NavbarRoutes/NavbarRoutes";
import SocialNetwork from "../SocialNetwork/SocialNetwork";
import { MainToolbarStyles } from "./MainToolbar.style";

export const MainToolbar = () => {
  const { isEqualOrLessThanMedium } = useMediaQuery();
  const [showMenu, setShowMenu] = useState(false);
  /*istanbul ignore next */
  const toogleMenu = () => setShowMenu(!showMenu);

  useEffect(() => {
    /*istanbul ignore next */
    if (!isEqualOrLessThanMedium) {
      setShowMenu(false);
    }
  }, [isEqualOrLessThanMedium]);

  return (
    <>
      <MainToolbarStyles.Toolbar>
        <Flexbox justifyContent="space-between" alignItems="center">
          <Box>
            {isEqualOrLessThanMedium ? (
              /*istanbul ignore next */
              <Image
                src="./images/menu.svg"
                alt="menu"
                width={30}
                height={30}
                style={{ cursor: "pointer" }}
                onClick={toogleMenu}
              />
            ) : (
              <SocialNetwork />
            )}
          </Box>
          <Box>{!isEqualOrLessThanMedium && <NavbarRoutes />}</Box>
        </Flexbox>
      </MainToolbarStyles.Toolbar>
      {isEqualOrLessThanMedium && (
        /*istanbul ignore next */
        <MainToolbarStyles.MenuLateral isShow={showMenu}>
          <MainToolbarStyles.IconCloseWrapper>
            <Image
              src="./images/close.svg"
              alt="menu"
              width={30}
              height={30}
              style={{ cursor: "pointer" }}
              onClick={toogleMenu}
            />
          </MainToolbarStyles.IconCloseWrapper>
          <NavbarRoutes />
        </MainToolbarStyles.MenuLateral>
      )}
    </>
  );
};
