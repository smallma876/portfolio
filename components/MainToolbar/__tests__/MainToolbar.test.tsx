import { render } from "@testing-library/react";
import MainToolbar from "../MainToolbar";

const renderComponent = async () => {
  const { container } = await render(<MainToolbar />);

  return container;
};

describe("MainToolbar", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });
  it("should render match snapshot", () => {
    const container = renderComponent();
    expect(container).toMatchSnapshot();
  });

  it("should render with mobile size", () => {
    const container = renderComponent();
    expect(container).toMatchSnapshot();
  });
});
