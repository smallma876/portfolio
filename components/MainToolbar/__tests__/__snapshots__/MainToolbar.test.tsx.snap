// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainToolbar should render match snapshot 1`] = `Promise {}`;

exports[`MainToolbar should render with mobile size 1`] = `Promise {}`;
